### PR TITLE
fix network data version and stable version issue

### PR DESCRIPTION
**Pull request by librasungirl**
change mVersion whenever NetworkData changes
change mStableVersion only when Stable NetworkData changes

@jwhui Please help to review, thanks!

### CODE REVIEW

File: src/core/thread/network_data_leader.cpp [hunk 1]
```diff
@@ -577,20 +575,200 @@ void Leader::SendServerDataResponse(const Coap::Header &aRequestHeader, const Ip
...
+    BorderRouterTlv *borderRouter;
+    HasRouteTlv *hasRoute;
+    BorderRouterEntry *borderRouterEntry;
+    HasRouteEntry *hasRouteEntry;
+    *aIn = false;
+    *aStable = false;
+
+    while (1)
+    {
+        if (cur >= end)
```
Comment (jwhui) [inline src/core/thread/network_data_leader.cpp]: Why not put the `cur >= end` in the `while` statement above and remove this `if` statement?

File: src/core/thread/network_data_leader.cpp [hunk 2]
```diff
@@ -577,20 +575,200 @@ void Leader::SendServerDataResponse(const Coap::Header &aRequestHeader, const Ip
...
+
+        if (cur->GetType() == NetworkDataTlv::kTypePrefix)
+        {
+            prefix = reinterpret_cast<PrefixTlv *>(cur);
+            subCur = reinterpret_cast<NetworkDataTlv *>(prefix->GetSubTlvs());
+            subEnd = reinterpret_cast<NetworkDataTlv *>(prefix->GetSubTlvs() + prefix->GetSubTlvsLength());
+
+            while (1)
+            {
+                if (subCur >= subEnd)
```
Comment (jwhui) [inline src/core/thread/network_data_leader.cpp]: Why not put `subCur < subEnd` in the `while` statement above and remove this `if` statement?

File: src/core/thread/network_data_leader.cpp [hunk 3]
```diff
@@ -577,20 +575,200 @@ void Leader::SendServerDataResponse(const Coap::Header &aRequestHeader, const Ip
     }
 }
 
+void Leader::RlocLookup(uint16_t aRloc16, bool *aIn, bool *aStable, uint8_t *aTlvs, uint8_t aTlvsLength)
```
Comment (jwhui) [inline src/core/thread/network_data_leader.cpp]: Use pass-by-reference for  `aIn` and `aStable`?

File: src/core/thread/network_data_leader.cpp [hunk 4]
```diff
@@ -577,20 +575,200 @@ void Leader::SendServerDataResponse(const Coap::Header &aRequestHeader, const Ip
...
+    NetworkDataTlv *cur = reinterpret_cast<NetworkDataTlv *>(aTlvs);
+    NetworkDataTlv *end = reinterpret_cast<NetworkDataTlv *>(aTlvs + aTlvsLength);
+    PrefixTlv *prefix;
+    PrefixTlv *prefixBase;
+    BorderRouterTlv *borderRouter;
+    HasRouteTlv *hasRoute;
+
+    while (1)
+    {
+        if (cur >= end)
```
Comment (jwhui) [inline src/core/thread/network_data_leader.cpp]: Why not put `cur < end` in the `while` statement above and remove this `if` statement?

Comment (jwhui) [inline src/core/thread/network_data_leader.cpp]: Why not put `cur < end` in the `while` statement above and remove this `if` statement?

File: src/core/thread/network_data_leader.cpp [hunk 5]
```diff
@@ -577,20 +575,200 @@ void Leader::SendServerDataResponse(const Coap::Header &aRequestHeader, const Ip
...
+    bool rlocStable = false;
+    bool stableUpdated = false;
 
-    mVersion++;
-    mStableVersion++;
+    RlocLookup(aRloc16, &rlocIn, &rlocStable, mTlvs, mLength);
+
+    if (aTlvs == NULL)
+    {
+        if (rlocIn)
```
Comment (jwhui) [inline src/core/thread/network_data_leader.cpp]: Can we move this code block up into `RemoveBorderRouter` (plus the few extra lines at the beginning and end)?  It's a little confusing to have `RegisterNetworkData` do two different things based on whether or not `aTlvs` is `NULL`.

**Comment by jwhui** [comment]
@librasungirl , thanks for submitting this fix.  I think this is pretty close.  See comments inline.  Also, we just enabled the code style checker, so make sure to run `make pretty` to make sure your code adheres to the code style.

**Comment by librasungirl** [comment]
hi @jwhui , when 'make pretty', error information would be as below
Invalid option file options:
style="allman"

it could work when remove the double quota around allman in the file .astyle-opts as below
--style=allman

**Comment by jwhui** [comment]
@librasungirl , create a separate GitHub issue for the astyle issue and indicate what environment (e.g. ubuntu/macosx/etc.) and version of astyle you are using.  Thanks.

**Pull request merged**